src/plugins/intel_gpu/src/kernel_selector/cl_kernels/softmax_gpu_bf.cl

```diff
@@ -93,14 +101,20 @@ KERNEL (softmax_gpu_continuous_bfyx)(
             BLOCK_TYPE_OPT vec_tmp = BLOCK_READ_OPT(input, aligned_data_offset + input_idx * get_sub_group_size());
             unroll_for (int j = 0; j < OPT_BLOCK_SIZE; j++)
             {
-                my_chunk[input_idx+j] = vec_tmp[j];
+                if (use_output_buffer)
+                    output[aligned_data_offset + get_sub_group_local_id() + (input_idx + j) * get_sub_group_size()] = vec_tmp[j];
+                else
+                    my_chunk[input_idx+j] = vec_tmp[j];
             }
         }
 
         for (; input_idx < items_num; input_idx++)
         {
             BLOCK_TYPE vec_tmp = BLOCK_READ(input, aligned_data_offset + input_idx * get_sub_group_size());
-            my_chunk[input_idx] = vec_tmp;
+            if (use_output_buffer)
+                output[aligned_data_offset + get_sub_group_local_id() + input_idx * get_sub_group_size()] = vec_tmp;
+            else
+                my_chunk[input_idx] = vec_tmp;
```

Comment:
random spot : 
Having branch within loop is not good for performance. 
I think you can take either of the approaches: 
(1) Using single kernel : 
- at the beginning, set the memory pointer like OUTPUT_TYPE* out_ptr = &my_chunk[0] or out_ptr = output; 
- likewise for the offsets and strides. 
(2) Prepare multiple kernels (kernel using slm && kernel not using slm) and select at runtime w.r.t the required size as it is done for fc and gemm

Reply:
Updated to use single if() when start.
Implementation of the 1st approach would not be feasible because available output buffer index should be calculated.
Let me measure performance in MTL with updated kernel first and will implement 2nd approach if performance drop is significant.